universal-application-tool-0.0.1/app/services/question/QuestionServiceImpl.java

```diff
@@ -129,6 +131,36 @@ public void discardDraft(Long id) throws InvalidUpdateException {
       throw new InvalidUpdateException("Did not find question in draft version.");
     }
     question.get().save();
+    versionRepositoryProvider.get().updateProgramsForNewDraftQuestion(id);
+  }
+
+  @Override
+  public ImmutableList<QuestionDefinition> getQuestionsForTag(QuestionTag tag) {
+    return questionRepository.getAllQuestionsForTag(tag);
+  }
+
+  @Override
+  public void setExportState(QuestionDefinition questionDefinition, String questionExportState)
+      throws QuestionNotFoundException, InvalidUpdateException {
+    Optional<Question> question =
+        questionRepository.lookupQuestion(questionDefinition.getId()).toCompletableFuture().join();
+    if (question.isEmpty()) {
+      throw new QuestionNotFoundException(questionDefinition.getId());
+    }
+    if (questionExportState.equals(QuestionTag.DEMOGRAPHIC.getValue())) {
+      question.get().removeTag(QuestionTag.DEMOGRAPHIC_PII);
+      question.get().addTag(QuestionTag.DEMOGRAPHIC);
+    } else if (questionExportState.equals(QuestionTag.DEMOGRAPHIC_PII.getValue())) {
+      question.get().removeTag(QuestionTag.DEMOGRAPHIC);
+      question.get().addTag(QuestionTag.DEMOGRAPHIC_PII);
+    } else if (questionExportState.equals("none")) {
```

Comment:
issue (non-blocking): this string shows up in a couple places, I'd prefer to at least see it in a constant. I can see how it doesn't exactly make sense to add it as a `QuestionTag` since it's the absence of a value rather. OTOH `QuestionTag.NON_DEMOGRAPHIC` seems less error prone?

Reply:
Yeah - fair enough.  Done.